model.go

```diff
@@ -118,21 +121,13 @@ func (m *Model) TableName() string {
 		return n.TableName(m.ctx)
 	}
 
+	m.isSlice()
+
 	if m.tableName != "" {
 		return m.tableName
 	}
 
-	t := reflect.TypeOf(m.Value)
-	name, cacheKey := m.typeName(t)
-
-	defer tableMapMu.Unlock()
-	tableMapMu.Lock()
-
-	if tableMap[cacheKey] == "" {
-		m.tableName = nflect.Tableize(name)
-		tableMap[cacheKey] = m.tableName
-	}
-	return tableMap[cacheKey]
```

Comment:
@stanislas-m the cache was completely ineffective. The only thing that was cached was the `m.tableName = nflect.Tableize(name)` statement here.

This does not reflect any value in my opinion as `nflect.Tableize()` is only doing some simple string parsing and concatenation.

Instead, this caused several hard to find bugs, and introduced a global lock which might even cause lock contestion and congestion in high traffic envs.

Reply:
Sure, dropping this cache makes sense. :)